packages/aws-cdk-lib/aws-applicationautoscaling/lib/step-scaling-policy.ts

```diff
@@ -111,6 +113,10 @@ export class StepScalingPolicy extends Construct {
       throw new Error('You must supply at least 2 intervals for autoscaling');
     }
 
+    if (props.scalingSteps.length > 40) {
+      throw new Error('You must have less than 40 steps in StepScalingPolicyProps.scalingSteps');
```

Comment:
should mention how many steps were requested

Reply:
Done.